packages/react-components/react-avatar/src/components/AvatarGroupOverflow/useAvatarGroupOverflow.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import { AvatarGroupContext } from '../../contexts/AvatarGroupContext';
+import { defaultAvatarGroupSize } from '../AvatarGroup/useAvatarGroup';
+import { resolveShorthand } from '@fluentui/react-utilities';
+import { MoreHorizontalRegular } from '@fluentui/react-icons';
+import { OnOpenChangeData, OpenPopoverEvents, Popover, PopoverSurface } from '@fluentui/react-popover';
+import { useContextSelector } from '@fluentui/react-context-selector';
+import type { AvatarGroupOverflowProps, AvatarGroupOverflowState } from './AvatarGroupOverflow.types';
+
+/**
+ * Create the state required to render AvatarGroupOverflow.
+ *
+ * The returned state can be modified with hooks such as useAvatarGroupOverflowStyles_unstable,
+ * before being passed to renderAvatarGroupOverflow_unstable.
+ *
+ * @param props - props from this instance of AvatarGroupOverflow
+ * @param ref - reference to root HTMLElement of AvatarGroupOverflow
+ */
+export const useAvatarGroupOverflow_unstable = (props: AvatarGroupOverflowProps): AvatarGroupOverflowState => {
+  const groupSize = useContextSelector(AvatarGroupContext, ctx => ctx.size);
```

Comment:
(nit) could skip creating an extra groupSize const:
```suggestion
  const size = useContextSelector(AvatarGroupContext, ctx => ctx.size) ?? defaultAvatarGroupSize;
```